presto-main/src/test/java/com/facebook/presto/sql/query/TestCurrentPath.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.query;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.sql.SqlPath;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+
+public class TestCurrentPath
+{
+    @Test
+    public void testCurrentUser()
+    {
+        Session session = testSessionBuilder().setPath(new SqlPath(Optional.of("testPath"))).build();
+        try (QueryAssertions queryAssertions = new QueryAssertions(session)) {
+            queryAssertions.assertQuery("SELECT CURRENT_PATH", "SELECT CAST('" + session.getPath().toString() + "' AS VARCHAR)");
+        }
```

Comment:
Would it make sense to add tests for null path and invalid paths?

Reply:
Setting an invalid path will result in a parsing error, so there's no way to ever have CURRENT_PATH to resolve to one.